### DISCUSSION TITLE
JSON Format for LangGraph .invoke()

### DISCUSSION THREAD

**Discussion by brian-nowak** [JoshuaC215/agent-service-toolkit#85] [Q&A]
Hi there! @JoshuaC215 this is a great project and is very helpful to see a full implementation of a LangGraph agent. 

I am trying to do something very similar, working within the constraints of Databricks. I want to use their new serving feature, [Apps](https://docs.databricks.com/en/dev-tools/databricks-apps/index.html) - which supports Streamlit projects, and models (incl. LangGraphs) served in Databricks.

The blocker I have right now is, using the model serving endpoint I need to pass a JSON payload to the endpoint, but I can't figure out how to do this with the thread_id required by LG for its MemorySaver checkpointer. 

Finally coming to my question =) - as you implemented the API piece, did you ever come across how one could pass the raw JSON into the model with a thread_id and get it to work with the MemorySaver? Put another way, when we do `app.invoke({"messages": messages}, config)` - is there a JSON representation of messages + config that you are aware of?

**Comment by JoshuaC215** [comment]
Hey, nice! I haven’t looked at this and don’t have access to a Databricks environment.

It sounds like this might be a general LangGraph issue and thus get a better review in their discussions on the main repo?

Is there some documentation about specifically running LangGraph on the databricks model / apps endpoint? That might be a good starting point. I’m curious about this capability too…

**Reply by brian-nowak** [reply]
Thanks for the reply! Yeah I was just curious if you had maybe tried something like that while serving up your graph with memory but I can see why you wouldn't have to worry about the raw payload using FastAPI and the built in LangGraph + Streamlit functions. 

It's tricky to search around for this because it seems like a minor detail, but I think enabling the memory along w/ a Databricks endpoint actually requires creating the chain/graph differently from the start. Trying to dig into some of the LangGraph tutorials to see if I uncover the fix. But appreciate the quick reply, and have a nice evening!
